include/arch/common/abstract-instruction-node.hpp

```diff
@@ -39,8 +40,19 @@ class AbstractInstructionNode : public AbstractSyntaxTreeNode
     /** \copydoc AbstractSyntaxTreeNode::getIdentifier() */
     const std::string& getIdentifier() const override;
 
+    /**
+     * Returns a Translateable text containing any help text/instruction documentation about this instruction.
+     * Note that this member function should be independent from any children as it may be called for the user
+     * to display information how to define a valid instruction of this kind.
+     * \return A Translateable text containing information dedicated to the user about this instruction
+     */
+    virtual const Translateable& getInstructionDocumentation() const = 0;
+
 protected:
 
+    /**
+     * Returns the InstructionInformation of this Instruction
```

Comment:
Prefer to write `\return the InstructionInformation ...` Because then it gets rendered better by doxygen